apps/landing-page/pages/blog/using-openshift-authentication-to-secure-access-to-backstage.mdx

```diff
@@ -0,0 +1,177 @@
+
+This capability can be enabled by setting the following Helm values.
+
+```yaml
+serviceAccount:
+  create: true
+  annotations:
+    serviceaccounts.openshift.io/oauth-redirecturi.backstage: "https://{{ .Values.ingress.host }}/oauth/callback"
+```
+
+### TLS Certificate Management
+
+There are multiple methods for which TLS certificates can be configured within OpenShift. They could be provided by the end user or by way of an operator, such as [cert-manager](https://docs.openshift.com/container-platform/4.12/security/cert_manager_operator/cert-manager-operator-install.html), which can integrate with an external certificate management tool or generate certificates of its own. OpenShift also includes support for automatically generating and injecting certificates into applications through the [Service Serving Certificate](https://docs.openshift.com/container-platform/4.12/security/certificates/service-serving-certificate.html) feature. The `service-ca` monitors annotations placed upon OpenShift `Service` resources. When a Service with the annotation `service.beta.openshift.io/serving-cert-secret-name=&lt;name>` is created, the controller generates a certificate and its associated private key within a `Secret`.
+
+To have OpenShift generate certificate within a Secret called `backstage-tls` which can be configured within the OAuth proxy, the following Helm values can be specified:
+
+```yaml
+service:
+  annotations:
+    service.alpha.openshift.io/serving-cert-secret-name: backstage-tls
+```
+
+### Ingress Configuration
+
+The ability to expose applications running within OpenShift externally has been one of the most compelling features included within OpenShift ever since the early days of version 3. Routes, the predecessor to the native Kubernetes Ingress resource, has enabled such functionality and it continues to this day. However, more and more applications are favoring the native Kubernetes Ingress option over the OpenShift specific Route.
+
+Fortunately, OpenShift can automatically “upconvert” Kubernetes native Ingress resources to OpenShift Routes automatically. In addition, the upconversion can be customized, and in particular, the TLS termination type to configure end-to-end secure transport. As described in the prior section, the OAuth proxy is secured using a certificate provided by the Service Serving Certificate feature. While this certificate is generated by a Certificate Authority that is trusted within OpenShift, end users would not be able to trust such certificates.
+
+To work around this challenge, the generated Route that is created from an Ingress resource can be configured with a TLS termination type of “reencrypt”. TLS communication is terminated at the Ingress router and re-encrypted for transport to the underlying pod. Since OpenShift assets trust the certificate that is used by the service-ca controller, trust is established for the final leg of communication enabling a fully trusted path from client to server.
```

Comment:
```suggestion
To work around this challenge, the generated `Route` that is created from an `Ingress` resource can be configured with a TLS termination type of "reencrypt". TLS communication is terminated at the Ingress router and re-encrypted for transport to the underlying pod. Since OpenShift assets trust the certificate that is used by the service-ca controller, trust is established for the final leg of communication enabling a fully trusted path from client to server.
```

`Since OpenShift assets trust the certificate that is used by ...` is that only my poor English or are there some words extra? `assets trust the certificate` doesn't sound right to me.

Reply:
I think it's fine!